examples/tutorials/tutor.py

```diff
@@ -12,1692 +12,45 @@
-        """
-        next = None
-        section = self.section
-        if len(section.subsections) > 0:
-            next = section.subsections[0]
-        else:
-            parent = section.parent
-            while parent is not None:
-                index = parent.subsections.index(section)
-                if index < (len(parent.subsections) - 1):
-                    next = parent.subsections[index + 1]
-                    break
-
-                parent, section = parent.parent, parent
-
-        return next
-
-    @cached_property
-    def _get_previous_section(self):
-        """ Returns the previous section of the tutorial.
-        """
-        previous = None
-        section = self.section
-        parent = section.parent
-        if parent is not None:
-            index = parent.subsections.index(section)
-            if index > 0:
-                previous = parent.subsections[index - 1]
-                while len(previous.subsections) > 0:
-                    previous = previous.subsections[-1]
-            else:
-                previous = parent
-
-        return previous
-
-    def _get_title(self):
-        """ Returns the title of the current section.
-        """
-        section = self.section
-        if section is None:
-            return ""
-
-        return "%s: %s" % (section.__class__.__name__, section.title)
-
-    def init_tutor(self):
-        """ Initials the tutor by creating the root section from the specified
-            path.
-        """
-        path = self.path
-        title = title_for(os.path.splitext(os.path.basename(path))[0])
-        section = SectionFactory(title=title).trait_set(path=path).section
-        if section is not None:
-            self.section = self.root = section
 
+def main(root_dir):
+    # Create a tutor and display the tutorial:
+    path, name = os.path.splitext(root_dir)
+    demo(dir_name=root_dir, title='Traits Demos')
 
-#  Run the program:
 
-# Only run the program if we were invoked from the command line:
 if __name__ == "__main__":
 
     # Validate the command line arguments:
     if len(sys.argv) > 2:
-        print(Usage)
+        print(usage)
         sys.exit(1)
 
     # Determine the root path to use for the tutorial files:
     if len(sys.argv) == 2:
-        path = sys.argv[1]
-    else:
-        path = os.getcwd()
-
-    # Create a tutor and display the tutorial:
-    tutor = Tutor(home=os.path.dirname(sys.argv[0])).trait_set(path=path)
-    if tutor.root is not None:
-        tutor.configure_traits()
+        root_dir = sys.argv[1]
     else:
-        print(
-            """No traits tutorial found in %s.
+        root_dir = os.getcwd()
 
-Correct usage is: python tutor.py [tutorial_path]
-where: tutorial_path = Path to the root of the traits tutorial.
-
-If tutorial_path is omitted, the current directory is assumed to be the root of
-the tutorial."""
-            % path
-        )
+    try:
+        main(root_dir)
+    except NameError as e:
```

Comment:
Not a review, just a question: Any idea why this `NameError` is needed? I see it is there in traitsui's tutor.py as well. It might be useful to comment/reference an issue if someone knows...

Reply:
Agreed that if this is necessary, we need a comment explaining why. And an issue too, if there's hope of eventually removing this.

Reply:
I'll actually remove it now and open an issue in TraitsUI - previous implementation (https://github.com/enthought/traitsui/pull/697) raised a `NameError` in `main()` if "no tutorials were found". Current implementation (https://github.com/enthought/traitsui/pull/712) will open any folder and attempt to show the contents, no error is raised.

Reply:
That's great. Thank you Ieva!

Reply:
Opened https://github.com/enthought/traitsui/issues/793